Libplanet.Tests/Store/Trie/MerkleTrieTest.cs

```diff
@@ -345,5 +346,137 @@ public void SetValueToFullNode()
             Assert.Equal(value0000, trie.Get(key0000));
             Assert.Equal(value0010, trie.Get(key0010));
         }
+
+        [Fact]
+        public void RemoveValue()
```

Comment:
IMO we were supposed to check `ITrie.Remove` is removed from `IKeyValueStore`. And I might we can check it with [`MemoryKeyValueStore`](https://github.com/planetarium/libplanet/blob/81408bf2f1fdce9f3674009438e2a038322ec2b6/Libplanet.Store/Trie/MemoryKeyValueStore.cs#L63). Can you add this test for sure?

Reply:
No, `ITrie.Remove()` should not remove the key value pair associated with the `KeyByte` being removed from `ITrie`. `IKeyValueStore.Delete()` is there for pruning. I suppose I can check that the key value pairs are **not** removed from the underlying `IKeyValueStore`. 😶

Reply:
See 4a3e409427c33b1fc7fce2180b16c607752645af.

Reply:
Hmmm... But you said when it commited, node is really remove from storage. right? https://github.com/planetarium/libplanet/pull/3576/files#diff-5050f0558f9c6e24f3d78bd6b017cb2bd65599e54280fa605d396f99cfcfbfc1R67-R68

Reply:
I guess its really the problem of wording I suppose. It isn't removed from the storage in the sense that there is no `ITrie` reference recorded in the storage with a value removed at given path. That is, it is "removed" only in the sense that you get an `ITrie` reference (i.e. `HashDigest<SHA256>`) that is "recorded" in the storage. I'll try to rephrase that bit. 🙄

Reply:
Tried to clarify a bit. See d1069af3da8589663bf448f6026ed0118613ec7a.

Reply:
Ok, then have you any idea for removing a dangling node from storage?

Reply:
It has to be manually removed? Depends on what we mean by dangling nodes. Technically, there shouldn't be any dangling nodes since all commits are used for `TxExecution` now. 🙄

Reply:
Oh, ok. I think I'm sure about that rn... LGTM.